scthumbd.js

```diff
@@ -57,18 +61,39 @@ const thumber = scThumber({
 const workers = process.env.WORKERS ?? os.cpus().length;
 const port = process.env.PORT ?? 4001;
 
-if (cluster.isPrimary && workers > 1) {
-  console.log(`${'[m]'.red} ${'scthumbd %s'.yellow}`, process.env.npm_package_version);
+if (cluster.isPrimary) {
+  const { version } = JSON.parse(fs.readFileSync('./package.json').toString());
+  console.log(`${'[m]'.red} ${'scthumbd %s'.yellow}`, version);
   console.log(`${'[m]'.red} Listening on port ${'%s'.green}...`, port);
-  console.log(`${'[m]'.red} Starting ${'%s'.green} workers...`, workers);
+}
 
+if (cluster.isPrimary && workers > 1) {
+  console.log(`${'[m]'.red} Starting ${'%s'.green} workers...`, workers);
   for (let i = 0; i < workers; i++) {
     cluster.fork();
   }
 
+  let shuttingDown = false;
+  for (const signal of ['SIGINT', 'SIGTERM']) {
+    process.on(signal, () => {
+      if (!shuttingDown) {
+        shuttingDown = true;
+        for (const worker of Object.values(cluster.workers)) {
+          if (!worker.isDead()) {
+            worker.send(SHUTDOWN_SIGNAL);
+          }
+        }
```

Comment:
just forward the signal to the workers as is?
```suggestion
      shuttingDown = true;
      for (const worker of Object.values(cluster.workers)) {
        worker.kill(signal);
      }
```

Reply:
This is what I was looking for. XD